ovos_phal_plugin_hotkeys/__init__.py

```
@@ -11,20 +11,89 @@
         super().__init__(bus=bus, name="ovos-PHAL-plugin-hotkeys", config=config)
+        # Assign a value to 'autoconfig_file'
+        # Can either contain a single line with values from 'ovos-i2csound',
+        # Or can be valid json with 'key_up' and or 'key_down'
+        self.autoconfig_file = self.config.get(
+            "autoconfig_file", "/home/ovos/.config/mycroft/i2c_platform")
+        # Check for existing configuration of 'key_up' and 'key_down'
+        if "key_up" in self.config or "key_down" in self.config:
+            # User has predefined settings.  Check for override
+            if self.config.get("autoconfigure"):
+                # User wants to override personal settings
+                self.autoconfig(self.autoconfig_file)
+        else:
+            # No configuration given, try to autodetect
+            LOG.debug(
+                f"no configuration given.  Trying to autodetect from {self.autoconfig_file}")
+            self.autoconfig(self.autoconfig_file)
+
         self.register_callbacks()
 
+    def autoconfig(self, config_file):
+        # Make sure the file exists
+        if exists(config_file):
+            # Check for json
+            try:
+                with open(config_file) as config:
+                    LOG.debug(f"loading json from {config_file}")
+                    new_config = json.load(config)
+                    self.load_config(new_config, json=True)
+            except JSONDecodeError:
+                with open(config_file, "r") as config:
+                    LOG.debug(f"file {config_file} is not json")
+                    platform = config.readline().strip()
+                    if platform in KNOWN_BOARDS:
+                        LOG.debug(f"loading config for {platform}")
+                        self.load_config(platform)
+            except Exception as e:
+                LOG.debug(e)
+        else:
+            LOG.debug(f"config file {config_file} does not exist")
```

i think this should throw an error and fail to load the plugin here